test/simulator/timestamp/src/timestamp_manager.cpp

```diff
@@ -222,3 +222,68 @@ timestamp_manager::validate_read_timestamp(session_simulator *session, const uin
 
     return (0);
 }
+
+/* Validate the commit timestamp. */
+int
+timestamp_manager::validate_commit_timestamp(
+  session_simulator *session, const uint64_t commit_ts) const
+{
+    uint64_t prepare_ts = session->get_prepare_timestamp();
+    if (!prepare_ts) {
```

Comment:
The below condition is true irrespective of prepare, so the validation should not be under the if condition.

Reply:
Can a prepared transaction set the commit timestamp multiple times?

Reply:
it doesn't matter as the commit timestamp can be set only after the prepare.

Reply:
The docs state: 

> For transactions that set multiple commit timestamps, the first commit timestamp set is also required to be the earliest: the second and subsequent commit timestamps may not be earlier than the first commit timestamp. This feature is not compatible with prepared transactions, which must use only a single commit timestamp.

However, I tried setting the commit_timestamp multiple times after preparing a transaction in a python example and WiredTiger doesn't complain. I don't think only being able to set the commit_timestamp once for prepared transactions is enforced?

Reply:
Lets move it out for readability sake, as discussed @Sean04